packages/runtime/lib/runtime.js

```diff
@@ -567,87 +563,31 @@ class Runtime extends EventEmitter {
     }, COLLECT_METRICS_TIMEOUT).unref()
   }
 
-  async pipeLogsStream (writableStream, logger, startLogId, endLogId, runtimePID) {
-    endLogId = endLogId || Infinity
-    runtimePID = runtimePID ?? process.pid
+  async addLoggerDestination (writableStream) {
+    // Add the stream - We output everything we get
+    this.#loggerDestination.add({ stream: writableStream, level: 1 })
 
-    const runtimeLogFiles = await this.#getRuntimeLogFiles(runtimePID)
+    // Immediately get the counter of the last add stream (writableStream) so we can use it to later remove it
+    const id = this.#loggerDestination.streams.reduce(function (accu, current) {
+      return current.id > accu ? current.id : accu
+    }, 0)
 
-    if (runtimeLogFiles.length === 0) {
-      writableStream.end()
-      return
-    }
-
-    let latestFileId = parseInt(runtimeLogFiles.at(-1).slice('logs.'.length))
-
-    let fileStream = null
-    let fileId = startLogId ?? latestFileId
-    let isClosed = false
-
-    const runtimeLogsDir = this.#getRuntimeLogsDir(runtimePID)
-
-    const watcher = watch(runtimeLogsDir, async (event, filename) => {
-      if (event === 'rename' && filename.startsWith('logs')) {
-        const logFileId = parseInt(filename.slice('logs.'.length))
-        if (logFileId > latestFileId) {
-          latestFileId = logFileId
-          fileStream.unwatch()
-        }
-      }
-    }).unref()
+    const onClose = () => {
+      writableStream.removeListener('close', onClose)
+      writableStream.removeListener('error', onClose)
+      this.removeListener('closed', onClose)
 
-    const streamLogFile = () => {
-      if (fileId > endLogId) {
-        writableStream.end()
-        return
+      if (!writableStream.destroyed) {
+        writableStream.destroy()
       }
 
-      const fileName = 'logs.' + fileId
-      const filePath = join(runtimeLogsDir, fileName)
-
-      const prevFileStream = fileStream
-
-      fileStream = ts.createReadStream(filePath)
-      fileStream.pipe(writableStream, { end: false, persistent: false })
+      const writableStreamIndex = this.#loggerDestination.streams.findIndex(s => s.id === id)
```

Comment:
Basically this implementing the remove in multistream?

1. open a pr to pino with the change
2. add a TODO here

Reply:
Pino PR: https://github.com/pinojs/pino/pull/2257